### DISCUSSION TITLE
Oauth backend extending extra_scope_source

### DISCUSSION THREAD

**Discussion by brunomedeirosdedalus** [rabbitmq/rabbitmq-server#7192] [Ideas]
Hi all, 

Maybe I am wrong, but as far as I know 'extra_scope_source' only allows  to be set to simple strings. Would it be worth it to extend this functionality to allow basic nested structure for example (from Keycloak) `realm_access.roles` this would allow mapping the RabbitMQ `scopes` to keycloak realm roles. The JWT structure:

```
  "realm_access": {
    "roles": [
      "lol",
  },
```

**Comment by michaelklishin** [comment]
On the Keycloak-specific code path, perhaps. I don't know if every Keycloak user would want that behavior, though.

**Reply by brunomedeirosdedalus** [reply]
Yes, that is true; I use it more as an example

**Reply by brunomedeirosdedalus** [reply]
The Keycloak was an example, the idea is to build something more generic to avoid:

```
    Payload2 = case maps:is_key(<<"authorization">>, Payload1) of
        true  -> post_process_payload_in_keycloak_format(Payload1);
```

If it is expressive enough then we can avoid having to explicitly dealt with cases like the aforementioned. For example, the user could map the extra_scope_source to 'authorization.permissions' and would automatically work for the aforementioned use case.

Just an idea, I would not mind spending something time and contributing with that if you feel that it is worth it.
